### DISCUSSION TITLE
Refresh Token Handling

### DISCUSSION THREAD

**Discussion by madc** [Diizzayy/nuxt-graphql-client#330] [Q&A]
I know, that there is a [closed issue (#12)](https://github.com/Diizzayy/nuxt-graphql-client/issues/12) discussing this topic, but I feel its not resolved and we are still struggling.

We are using Directus as a backend and for [authentication](https://docs.directus.io/reference/authentication.html). Upon login we receive `access_token` and `refresh_token`. Once the `access_token` expires, any API call fails with `TOKEN_EXPIRED` and we are supposed to renew it with the `refresh_token`. So far we can handle this in a plugin using `useGqlError`.

**The issue we are facing**: There seems to be no way to redo the initial API call, that failed with `TOKEN_EXPIRED`. Ideally, the GraphQL client would redo this call and return the result.

Has anyone resolved this successfully?
Thanks for any feedback!

**Comment by BorisKamp** [comment]
Plus one!
